src/py/aspen/covidhub_import/implementation.py

```diff
@@ -0,0 +1,136 @@
+            .one_or_none()
+        )
+        if group is None:
+            # copy the project info into the "group"
+            group = Group(
+                name=project.originating_lab,
+                address=project.originating_address,
+                # FIXME: this needs to be updated.
+                email="FAKE_EMAIL@gmail.com",
+            )
+            session.add(group)
+
+        consensus_genomes: Iterable[ConsensusGenome] = (
+            covidhub_session.query(ConsensusGenome)
+            .join(SampleFastqs, CZBID, Project)
+            .distinct(CZBID.id)
+            .filter(Project.rr_project_id == rr_project_id)
+            .filter(
+                or_(
+                    ConsensusGenome.status == ConsensusGenomeStatus.PASSED_AUTO,
+                    ConsensusGenome.status == ConsensusGenomeStatus.PASSED_MANUAL,
+                )
+            )
+            .order_by(
+                CZBID.id.desc(),
+                ConsensusGenome.recovered_sites.desc(),
+                ConsensusGenome.id.desc(),
+            )
+            .options(
+                joinedload(ConsensusGenome.sample_fastqs)
+                .joinedload(SampleFastqs.czb_id)
+                .joinedload(CZBID.genome_submission_info),
+            )
+        )
+
+        external_accessions = {
+            consensus_genome.sample_fastqs.czb_id.external_accession
+            for consensus_genome in consensus_genomes
+            if isinstance(consensus_genome.sample_fastqs.czb_id, DphCZBID)
+        }
+        external_accessions_to_samples: Mapping[str, Sample] = {
+            sample.private_identifier: sample
+            for sample in session.query(Sample).filter(
+                and_(
+                    Sample.submitting_group == group,
+                    Sample.private_identifier.in_(external_accessions),
+                )
+            )
+        }
+
+        for consensus_genome in consensus_genomes:
+            czbid = consensus_genome.sample_fastqs.czb_id
+            if not isinstance(czbid, DphCZBID):
+                warnings.warn(
+                    "consensus genome has non-dphczbid"
+                    f"{consensus_genome.sample_fastqs.czb_id}"
+                )
+                continue
+            external_accession = czbid.external_accession
+            sample = external_accessions_to_samples.get(external_accession, None)
+            if sample is None:
+                sample = Sample(
+                    submitting_group=group, private_identifier=external_accession
+                )
+
+            sample.original_submission = {}
+            sample.public_identifier = (
+                f"USA/{czbid.submission_base}/{czbid.collection_date.year}"
+            )
+            sample.sample_collected_by = project.originating_lab
+            sample.sample_collector_contact_address = project.originating_address
+            sample.collection_date = czbid.collection_date
```

Comment:
`collection_date` doesn't exist for `InternalCZBID`. I see you're outputting a warning for non-DPH samples, but it isn't clear to me whether you expect internal CZBIDs here. If you really need it, there's some complicated function to grab the date for internal czbids, though note it is bugged for samples that were PCR'd twice::
https://github.com/czbiohub/covidhub/issues/1485

Reply:
I think for Orange County, it's all DphCZBID.  I'll resolve when we get to that.

Reply:
https://app.clubhouse.io/genepi/story/123614/importing-covidhub-data-does-not-work-for-non-dphczbid